tests/test_compare_with_default_qubit.py

```diff
@@ -55,110 +54,83 @@ def setUp(self):
-                        elif observable_class.par_domain == 'A':
-                            if str(observable) == "Hermitian":
-                                observable_pars = [np.array([[1,1j],[-1j,0]])]
-                            else:
-                                raise IgnoreOperationException('Skipping in automatic test because I don\'t know how to generate parameters for the observable '+observable+" with par_domain="+str(observable_class.par_domain))
-                        elif str(observable) == "SparseHamiltonian":
-                            raise IgnoreOperationException(
-                                'Skipping in automatic test because' + observable + 'is not valid in ProjectQ.')
-                        else:
-                            observable_pars = {}
-
-                        # apply to the first wires
-                        operation_wires = list(range(operation_class.num_wires)) if operation_class.num_wires > 1 else 0
-                        observable_wires = list(range(observable_class.num_wires)) if observable_class.num_wires > 1 else 0
-
-                        if str(operation) == "QubitStateVector":
-                            operation_wires = range(self.num_subsystems)
-
-                        operation_class(*operation_pars, wires=operation_wires)
-                        return qml.expval(observable_class(*observable_pars, wires=observable_wires))
-
-                    try:
-                        output = circuit()
-                    except IgnoreOperationException:
-                        continue
-
-                    if (operation, observable) not in outputs:
-                        outputs[(operation, observable)] = {}
-
-                    outputs[(operation, observable)][str(type(dev).__name__)+"(shots="+str(dev.shots)+")"] = output
-
-        #if we could run the circuit on more than one device assert that both should have given the same output
-        for (key,val) in outputs.items():
-            if len(val) >= 2:
-                self.assertAllElementsAlmostEqual(val.values(), delta=self.tol, msg="Outputs "+str(list(val.values()))+" of devices ["+', '.join(list(val.keys()))+"] do not agree for a circuit consisting of a "+str(key[0])+" Operation followed by a "+str(key[1])+" Expectation." )
-
-
-if __name__ == '__main__':
-    log.info('Testing PennyLane ProjectQ Plugin version ' + qml.version() + ', Device class.')
-    # run the tests in this file
-    suite = unittest.TestSuite()
-    for t in (CompareWithDefaultQubitTest, ):
-        ttt = unittest.TestLoader().loadTestsFromTestCase(t)
-        suite.addTests(ttt)
-
-    unittest.TextTestRunner().run(suite)
+                        """4-qubit circuit with layers of randomly selected gates and random connections for
+                        multi-qubit gates."""
+                        qml.BasisState(np.array([1, 0, 0, 0]), wires=[0, 1, 2, 3])
+                        np.random.seed(1967)
```

Comment:
What does re-setting the seed here do? I think the `gates_per_layers` is already generated by this point, right? Or does it affect the circuit result in some way? (I would think in theory the expectation value should stay the same.)

Reply:
Good point let me check that!